### PR TITLE
Migrate use, use_index to ArgParse

### PR DESCRIPTION
**Pull request by t-wissmann**
This makes use of ArgParse in the commands 'use' and 'use_index', and
makes them part of the GlobalCommands class. They both need access to
TagManager and MonitorManager, so they don't really belong to neither of
those classes. Since they modify the global state, they fit quite well
in GlobalCommands.

For the implementation, TagManager now provides a run-time converter
for tags, but it only can be used for 'use'.

Also, this adds test cases for both commands and thus also for the
run-time converter.

### CODE REVIEW

**Comment by t-wissmann** [comment]
The tests should be fixed as soon as #1226 is merged.

File: tests/test_global_commands.py [hunk 1]
```diff
@@ -134,3 +137,83 @@ def test_cycle_value_invalid_arg(hlwm):
     # calling it a second time fails
     hlwm.call_xfail(command) \
         .expect_stderr('Invalid value for "clients.my_foo": ')
+
+
+def test_use_tag(hlwm):
```
Comment (blarz) [inline tests/test_global_commands.py:142]: There is a similar test in test_tags.py. Should we remove it?

**Review by blarz — CHANGES_REQUESTED** [review]

File: tests/test_global_commands.py (hunk 1 above)
Comment (t-wissmann) [reply tests/test_global_commands.py:142]: I've merged the test cases into one the test case `test_use_tag` in `test_global_commands.py`

Comment (blarz) [reply tests/test_global_commands.py:142]: Thanks!

**Review by blarz — APPROVED** [review]

**Pull request merged**